arangod/Cache/PlainCache.cpp

```diff
@@ -269,10 +269,10 @@ bool PlainCache<Hasher>::freeMemoryWhile(
       bucket.evict(candidate);
       freeValue(candidate);
       maybeMigrate |= guard.source()->slotEmptied();
-    }
 
-    if (!cb(reclaimed)) {
-      break;
+      if (!cb(reclaimed)) {
```

Comment:
What it `reclaimed == 0`? Is that possible?

Reply:
It's true, in theory `reclaimed` is not checked for `0` here.
However, we only get in here if `candidate != nullptr` and then we calculate `reclaimed` as `candidate->size()`.
And `CachedValue::size()` is defined as
```
inline std::size_t size() const noexcept {
 return _headerAllocSize + keySize() + valueSize();
}
```
(with `_headerAllocSize` being > 0).
So in practice it should not happen that `reclaimed` is `0` when we get here.